udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,36 +44,48 @@ private String arrayListToJson(ArrayList<String> inputList) {
   }
 
   /**
-  * Obtains comments json to display upon request.
-  */
+   * Obtains comments json to display upon request.
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
+    Query query = new Query("Comms").addSort("timestamp", SortDirection.DESCENDING);
+
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> comments = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String commentText = (String) entity.getProperty("text");
+      comments.add(commentText);
+    }
+
     String outputJson = arrayListToJson(comments);
+
     response.getWriter().println(outputJson);
   }
 
   /**
-  * Writes comment data to ArrayList data
-  */
+   * Writes comment data to ArrayList data
```

Comment:
Documentation needs to be updated (no longer writing to ArrayList)?

Reply:
Done